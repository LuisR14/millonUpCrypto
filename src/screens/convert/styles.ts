import {Dimensions, Platform, StyleSheet} from 'react-native';
import {dimension} from '../../themes/dimension';
import colors from '../../themes/colors';
import fontSize from '../../themes/fontSize';
const {horizontalScale, verticalScale} = dimension();
const {height} = Dimensions.get('window');

export default StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    marginHorizontal: 20,
  },
  textInputContainer: {
    height: verticalScale(70),
    marginTop: 20,
    borderRadius: 8,
    flexDirection: 'row',
    width: horizontalScale(330),
    paddingVertical: 10,
    paddingHorizontal: 10,
    alignItems: 'center',
    shadowOffset: {
      width: 0,
      height: 10,
    },
    shadowOpacity: 0.3,
    shadowRadius: 5,
    elevation: 10,
  },
  textInput: {
    flex: 1,
    paddingHorizontal: 10,
    paddingVertical: 5,
    color: colors.black,
    fontSize: fontSize.mlarge,
  },
  textInputDisable: {
    flex: 1,
    paddingHorizontal: 10,
    paddingVertical: 5,
    fontSize: fontSize.large,
  },
  textTitle: {
    fontWeight: 'bold',
    paddingHorizontal: 10,
    marginTop: 20,
    color: colors.black,
    fontSize: fontSize.xlarge,
  },
  dropDown: {
    borderColor: colors.transparent,
  },
  dropDownContainerStyle: {
    width: horizontalScale(140),
  },
  dropDownContainer: {
    width: horizontalScale(130),
    borderTopColor: colors.transparent,
    borderWidth: 0.3,
    paddingVertical: 5,
    borderColor: colors.gray,
    marginLeft: height > 700 ? -9.9 : -9.5,
    marginTop:
      Platform.OS === 'android'
        ? height > 740
          ? 10
          : 7
        : height > 670
        ? 12
        : 4,
  },
  dropDownItemContainer: {
    paddingVertical: 10,
  },
  dropDownItemSeparator: {
    backgroundColor: colors.gray,
    marginVertical: 5,
  },
  dropDownSearchContainer: {
    marginBottom: 5,
  },
  cryptoIcon: {
    height: 40,
    width: 40,
  },
});
