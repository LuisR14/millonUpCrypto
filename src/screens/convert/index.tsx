/* eslint-disable react-hooks/exhaustive-deps */
/* eslint-disable react/no-unstable-nested-components */
import {useRoute} from '@react-navigation/native';
import React, {useEffect, useState} from 'react';
import {Image, SafeAreaView, TouchableOpacity, View, Text} from 'react-native';
import styles from './styles';
import colors from '../../themes/colors';
import Margin from '../../components/margin';
import {useSelector} from 'react-redux';
import {CryptoProps} from '../../types/crypto';
import ChangeIcon from '../../assets/icons/change.svg';
import {CryptoSelectPicker} from '../../components/cryptoSelectPicker';

export const ConvertScreen = () => {
  const params = useRoute<any>();
  const cryptoValueParam = params?.params?.cryptoValue;
  const [cryptoFromValue, setcryptoFromValue] = useState('');
  const [items, setItems] = useState<any[]>([]);
  const [cryptoToValue, setcryptoToValue] = useState('');
  const [valueTo, setValueTo] = useState('1');
  const [valueFrom, setValueFrom] = useState(cryptoValueParam);
  const [openFrom, setOpenFrom] = useState(false);
  const [openTo, setOpenTo] = useState(false);
  const zIndex = 1;
  const borderBottomRadius = 0;
  const borderBottomRadiusClose = 8;
  const cryptoList = useSelector((state: any) => state.cryptoList);
  const Settings = useSelector((state: any) => state.userSettings);
  const Theme = Settings?.darkMode;

  useEffect(() => {
    if (cryptoList?.listBackup?.length > 0 && !cryptoValueParam) {
      setValueFrom(cryptoList?.listBackup[0].price_usd);
    }
  }, [cryptoList?.listBackup]);

  useEffect(() => {
    const cryptoListDropDown: React.SetStateAction<any[]> = [
      {
        id: 0,
        label: 'USD',
        value: '1',
        icon: () => (
          <Image
            source={{
              uri: 'https://coinicons-api.vercel.app/api/icon/usd',
            }}
            style={styles.cryptoIcon}
          />
        ),
      },
    ];
    cryptoList?.listBackup?.map((cryptoItem: CryptoProps) => {
      cryptoListDropDown.push({
        id: cryptoItem?.id,
        label: cryptoItem?.symbol,
        value: cryptoItem?.price_usd,
        icon: () => (
          <Image
            source={{
              uri: `https://coinicons-api.vercel.app/api/icon/${cryptoItem?.symbol?.toLowerCase()}`,
            }}
            style={styles.cryptoIcon}
          />
        ),
      });
    });
    setItems(cryptoListDropDown);
  }, []);

  useEffect(() => {
    const cryptoValueNumber = parseFloat(cryptoFromValue);
    const valueFromNumber = parseFloat(valueFrom);
    const valueToNumber = parseFloat(valueTo);
    if (cryptoFromValue) {
      if (valueToNumber === 1) {
        setcryptoToValue((cryptoValueNumber * valueFromNumber).toFixed(5));
      } else {
        setcryptoToValue(
          ((valueFromNumber / valueToNumber) * cryptoValueNumber).toFixed(5),
        );
      }
    } else {
      setcryptoToValue('0.00');
    }
  }, [cryptoFromValue, valueFrom, valueTo]);

  const onChangeOrder = () => {
    setValueFrom(valueTo);
    setValueTo(valueFrom);
  };

  return (
    <SafeAreaView testID="safeAreaViewTest" style={styles.container}>
      <Text
        testID="titleConvertScreenTest"
        style={{
          ...styles.textTitle,
          color: Theme !== 'dark' ? colors.black : colors.white,
        }}>
        Convert
      </Text>
      <View
        testID="viewTextInputTest"
        style={{
          ...styles.textInputContainer,
          zIndex,
          borderBottomLeftRadius: openFrom
            ? borderBottomRadius
            : borderBottomRadiusClose,
          backgroundColor: Theme !== 'dark' ? colors.lightGray : colors.gray,
        }}>
        <CryptoSelectPicker
          testID="selectPickerUp"
          editable={true}
          openFrom={openFrom}
          valueFrom={valueFrom}
          items={items}
          setOpenFrom={setOpenFrom}
          setValueFrom={setValueFrom}
          setItems={setItems}
          cryptoFromValue={cryptoFromValue}
          setcryptoFromValue={setcryptoFromValue}
        />
      </View>
      <Margin bottom={40} />
      <TouchableOpacity onPress={onChangeOrder}>
        <ChangeIcon fill={Theme !== 'dark' ? colors.black : colors.white} />
      </TouchableOpacity>
      <Margin bottom={20} />
      <View
        testID="viewTextInput2Test"
        style={{
          ...styles.textInputContainer,
          borderBottomLeftRadius: openTo
            ? borderBottomRadius
            : borderBottomRadiusClose,
          backgroundColor: Theme !== 'dark' ? colors.lightGray : colors.gray,
        }}>
        <CryptoSelectPicker
          testID="selectPickerDown"
          editable={false}
          openFrom={openTo}
          valueFrom={valueTo}
          items={items}
          setOpenFrom={setOpenTo}
          setValueFrom={setValueTo}
          setItems={setItems}
          cryptoFromValue={cryptoToValue}
        />
      </View>
    </SafeAreaView>
  );
};
