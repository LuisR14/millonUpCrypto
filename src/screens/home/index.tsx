/* eslint-disable react-hooks/exhaustive-deps */
import React, {useCallback, useEffect, useRef, useState} from 'react';
import {
  Modal,
  RefreshControl,
  Text,
  TextInput,
  TouchableHighlight,
  TouchableOpacity,
  View,
} from 'react-native';
import {CryptoProps} from '../../types/crypto';
import styles from './styles';
import {RenderFooter} from '../../components/flatlistFooter';
import CryptoComponent from '../../components/cryptoList';
import Margin from '../../components/margin';
import SearchIcon from '../../assets/icons/search.svg';
import CleanSearchIcon from '../../assets/icons/cleanSearch.svg';
import colors from '../../themes/colors';
import {getCryptoList} from '../../functions/getCryptoList';
import {loadMoreCrypto} from '../../functions/loadMoreCrypto';
import {searchText} from '../../functions/searchText';
import {SkeletonLoader} from '../../components/skeletonLoader';
import {ModalDetailComponent} from '../../components/modalDetail';
import {useDispatch, useSelector} from 'react-redux';
import {
  create,
  createBackup,
  update,
  updateBackup,
} from '../../redux/cryptoListSlice';
import {CryptoListEmpty} from '../../components/cryptoListEmpty';
import {sleep} from '../../functions/sleep';
import Animated, {
  useAnimatedScrollHandler,
  useSharedValue,
} from 'react-native-reanimated';
import {SafeAreaView} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {filter} from '../../functions/filterCryptoList';
import {saveCryptoList} from '../..//functions/saveCryptoList';
import {getCryptoListSaved} from '../..//functions/getCryptoListSaved';
import {useNetInfo} from '@react-native-community/netinfo';
import {useToast} from 'react-native-toast-notifications';

export const HomeScreen = () => {
  const [cryptoDetail, setcryptoDetail] = useState<CryptoProps>();
  const [isLoading, setisLoading] = useState(false);
  const [isRefreshing, setisRefreshing] = useState(false);
  const [isLoadingMore, setisLoadingMore] = useState(false);
  const [count, setcount] = useState(1);
  const [search, setsearch] = useState('');
  const [modalDetail, setmodalDetail] = useState(false);
  const [blockUpdate, setblockUpdate] = useState(true);
  const [activefilterGainer, setActivefilterGainer] = useState(false);
  const [activefilterLosers, setActivefilterLosers] = useState(false);
  const [activefilterVolume24, setActivefilterVolume24] = useState(false);
  const [errorMsg, setErrorMsg] = useState(null);
  const dispatch = useDispatch();
  const searchRef = useRef<any>();
  const flatListRef = useRef<any>();
  const cryptoList = useSelector((state: any) => state.cryptoList);
  const Settings = useSelector((state: any) => state.userSettings);
  const Theme = Settings?.darkMode;
  const netInfo = useNetInfo();
  const Offline = Settings?.offline;
  const scrollY = useSharedValue(0);
  const toast = useToast();
  useEffect(() => {
    SplashScreen?.hide();
    setisLoading(true);
  }, []);

  useEffect(() => {
    if (netInfo.isConnected !== null) {
      checkCryptoApi();
    }
  }, [netInfo.isConnected]);

  useEffect(() => {
    if (Offline) {
      setblockUpdate(true);
      getSavedCryptoList();
    } else if (!Offline && netInfo.isConnected !== null) {
      setblockUpdate(false);
      checkCryptoApi();
    }
  }, [Offline]);

  useEffect(() => {
    if (errorMsg) {
      toast.show(errorMsg, {
        type: 'danger',
      });
      setErrorMsg(null);
    }
  }, [toast, errorMsg]);

  const checkCryptoApi = async () => {
    cleanFilter();
    if (Offline || netInfo.isConnected === false) {
      getSavedCryptoList();
    } else {
      const cryptoListResponse = await getCryptoList({
        setsearch,
      });
      if (cryptoListResponse?.length > 0) {
        dispatch(create(cryptoListResponse));
        dispatch(createBackup(cryptoListResponse));
        saveCryptoList(cryptoListResponse);
        setblockUpdate(false);
      } else {
        setErrorMsg('server error');
      }
    }
    setisLoading(false);
  };
  const updateCryptoList = async () => {
    const cryptoListUpdate = await loadMoreCrypto({
      setisLoadingMore,
      count,
      setcount,
      search,
    });
    if (cryptoListUpdate.length > 0) {
      dispatch(update(cryptoListUpdate));
      dispatch(updateBackup(cryptoListUpdate));
    } else {
      setErrorMsg('server error');
    }
  };

  const getSavedCryptoList = async () => {
    const list = await getCryptoListSaved();
    dispatch(create(list));
    dispatch(createBackup(list));
  };

  const handler = useAnimatedScrollHandler({
    onScroll: event => {
      scrollY.value = event.contentOffset.y;
    },
  });

  const scollToTop = () => {
    flatListRef.current.scrollToOffset({
      animated: true,
      offset: 0,
    });
  };

  const cleanFilter = () => {
    setActivefilterGainer(false);
    setActivefilterLosers(false);
    setActivefilterVolume24(false);
  };

  const keyExtractor = useCallback((item: CryptoProps) => item.id, []);

  const renderItems = useCallback(
    ({item, index}: any) => (
      <TouchableHighlight
        testID="buttonDetailsTest"
        style={styles.touchable}
        activeOpacity={0.6}
        underlayColor="lightgray"
        onPress={() => {
          setcryptoDetail(item);
          setmodalDetail(true);
        }}>
        <CryptoComponent cryptoItem={item} scrollY={scrollY} index={index} />
      </TouchableHighlight>
    ),
    [],
  );

  return (
    <>
      <SafeAreaView style={styles.container}>
        {isLoading && !isRefreshing ? (
          <SkeletonLoader />
        ) : (
          <>
            <View testID="viewTextInputTest" style={styles.textInputContainer}>
              <TouchableOpacity
                style={styles.touchableIconSearch}
                onPress={() => searchRef.current.focus()}>
                <SearchIcon />
              </TouchableOpacity>
              <TextInput
                testID="textInputTest"
                ref={searchRef}
                style={styles.textInput}
                placeholder="Search coin"
                placeholderTextColor={colors.gray}
                value={search}
                onChangeText={text => {
                  scollToTop();
                  cleanFilter();
                  const cryptoListFilter = searchText({
                    text,
                    setsearch,
                    cryptoList,
                  });
                  if (cryptoListFilter) {
                    dispatch(create(cryptoListFilter));
                  } else {
                    dispatch(create(cryptoList?.listBackup));
                  }
                }}
              />
              {search.length > 0 && (
                <TouchableOpacity
                  style={styles.touchableIconSearch}
                  onPress={() => {
                    setsearch('');
                    dispatch(create(cryptoList?.listBackup));
                  }}>
                  <CleanSearchIcon />
                </TouchableOpacity>
              )}
            </View>
            <Margin bottom={15} />
            <View style={styles.filtersContainer}>
              <TouchableOpacity
                testID="buttonFilterGainers"
                onPress={() => {
                  scollToTop();
                  if (activefilterGainer) {
                    setblockUpdate(false);
                    dispatch(create(cryptoList?.listBackup));
                  } else {
                    setActivefilterLosers(false);
                    setActivefilterVolume24(false);
                    setblockUpdate(true);
                    const filterlist = filter(
                      cryptoList?.listBackup,
                      'gainers',
                    );
                    dispatch(create(filterlist));
                  }
                  setActivefilterGainer(!activefilterGainer);
                }}
                style={{
                  ...styles.filtersButton,
                  backgroundColor: activefilterGainer
                    ? Theme === 'dark'
                      ? colors.white
                      : colors.lightBlue
                    : Theme === 'dark'
                    ? colors.gray
                    : colors.lightGray,
                }}>
                <Text style={styles.textFilters}>Gainers</Text>
              </TouchableOpacity>
              <TouchableOpacity
                testID="buttonFilterLosers"
                onPress={() => {
                  scollToTop();
                  if (activefilterLosers) {
                    setblockUpdate(false);
                    dispatch(create(cryptoList?.listBackup));
                  } else {
                    setActivefilterGainer(false);
                    setActivefilterVolume24(false);
                    setblockUpdate(true);
                    const filterlist = filter(cryptoList?.listBackup, 'losers');
                    dispatch(create(filterlist));
                  }
                  setActivefilterLosers(!activefilterLosers);
                }}
                style={{
                  ...styles.filtersButton,
                  backgroundColor: activefilterLosers
                    ? Theme === 'dark'
                      ? colors.white
                      : colors.lightBlue
                    : Theme === 'dark'
                    ? colors.gray
                    : colors.lightGray,
                }}>
                <Text style={styles.textFilters}>Losers</Text>
              </TouchableOpacity>
              <TouchableOpacity
                testID="buttonFilterVolume24"
                onPress={() => {
                  scollToTop();
                  if (activefilterVolume24) {
                    setblockUpdate(false);
                    dispatch(create(cryptoList?.listBackup));
                  } else {
                    setActivefilterLosers(false);
                    setActivefilterGainer(false);
                    setblockUpdate(true);
                    const filterlist = filter(
                      cryptoList?.listBackup,
                      'gainers',
                    );
                    dispatch(create(filterlist));
                  }
                  setActivefilterVolume24(!activefilterVolume24);
                }}
                style={{
                  ...styles.filtersButton,
                  backgroundColor: activefilterVolume24
                    ? Theme === 'dark'
                      ? colors.white
                      : colors.lightBlue
                    : Theme === 'dark'
                    ? colors.gray
                    : colors.lightGray,
                }}>
                <Text style={styles.textFilters}>Volume24h</Text>
              </TouchableOpacity>
            </View>
            <Margin bottom={10} />
            <Animated.FlatList
              testID="flatListTest"
              ref={flatListRef}
              contentContainerStyle={styles.flatlistContainer}
              keyboardDismissMode="on-drag"
              ListEmptyComponent={CryptoListEmpty}
              onScroll={handler}
              maxToRenderPerBatch={40}
              onEndReached={async () => {
                if (!blockUpdate) {
                  setblockUpdate(true);
                  updateCryptoList();
                  await sleep(5000);
                  setblockUpdate(false);
                }
              }}
              onEndReachedThreshold={1}
              refreshControl={
                <RefreshControl
                  refreshing={isRefreshing}
                  onRefresh={() => {
                    setisRefreshing(true);
                    setTimeout(() => {
                      setisRefreshing(false);
                    }, 400);
                    checkCryptoApi();
                  }}
                  title="Refreshing"
                  tintColor={Theme === 'dark' ? colors.white : colors.black}
                  titleColor={Theme === 'dark' ? colors.white : colors.black}
                  colors={
                    Theme === 'dark' ? [colors.lightBlue] : [colors.black]
                  }
                />
              }
              data={cryptoList?.list}
              renderItem={renderItems}
              keyExtractor={keyExtractor}
              ListFooterComponent={() => RenderFooter(isLoadingMore)}
            />
            <Modal
              testID="modalTest"
              style={{
                backgroundColor:
                  Theme !== 'dark' ? colors.white : colors.darkCyan,
              }}
              animationType="slide"
              visible={modalDetail}>
              <ModalDetailComponent
                cryptoItem={cryptoDetail}
                setmodalDetail={setmodalDetail}
              />
            </Modal>
          </>
        )}
      </SafeAreaView>
    </>
  );
};
