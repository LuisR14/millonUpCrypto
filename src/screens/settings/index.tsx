import React, {useState} from 'react';
import {SafeAreaView, Switch, View} from 'react-native';
import {Text} from 'react-native';
import styles from './styles';
import Margin from '../../components/margin';
import {useDispatch, useSelector} from 'react-redux';
import {updateUserOffline, updateUserTheme} from '../../redux/userSettingSlice';
import {saveUserTheme} from '../../functions/saveUserTheme';
import colors from '../../themes/colors';
import {useNetInfo} from '@react-native-community/netinfo';

export const SettingsScreen = () => {
  const dispatch = useDispatch();
  const Settings = useSelector((state: any) => state.userSettings);
  const Theme = Settings?.darkMode;
  const Offline = Settings?.offline;
  const [switchValue, setswitchValue] = useState(
    Theme === 'dark' ? true : false,
  );
  const netInfo = useNetInfo();

  const onChangeSwitch = (value: boolean) => {
    if (value) {
      setswitchValue(true);
      dispatch(updateUserTheme('dark'));
      saveUserTheme('dark');
    } else {
      setswitchValue(false);
      dispatch(updateUserTheme('light'));
      saveUserTheme('light');
    }
  };
  const onChangeSwitchOffline = (value: boolean) => {
    if (value) {
      dispatch(updateUserOffline(value));
    } else {
      dispatch(updateUserOffline(value));
    }
  };

  return (
    <SafeAreaView testID="safeAreaViewTest" style={styles.container}>
      <Text
        testID="textTitleTest"
        style={{
          ...styles.textTitle,
          color: Theme !== 'dark' ? colors.black : colors.white,
        }}>
        Settings
      </Text>
      <Margin bottom={20} />
      <View
        style={{
          ...styles.settingComponent,
          backgroundColor: Theme !== 'dark' ? colors.lightBlue : colors.white,
        }}>
        <Text testID="darkModeTextSettingTest" style={styles.textSubTitle}>
          Dark mode
        </Text>
        <View testID="viewSwitch1Test" style={styles.switchContainer}>
          <Switch
            testID="switch1Test"
            onValueChange={value => onChangeSwitch(value)}
            value={switchValue}
          />
        </View>
      </View>
      <View
        style={{
          ...styles.settingComponent,
          backgroundColor: Theme !== 'dark' ? colors.lightBlue : colors.white,
        }}>
        <Text testID="offlineModeTextSettingTest" style={styles.textSubTitle}>
          Offline mode
        </Text>
        <View testID="viewSwitch2Test" style={styles.switchContainer}>
          <Switch
            testID="switch2Test"
            disabled={!netInfo.isConnected}
            onValueChange={value => onChangeSwitchOffline(value)}
            value={Offline}
          />
        </View>
      </View>
    </SafeAreaView>
  );
};
