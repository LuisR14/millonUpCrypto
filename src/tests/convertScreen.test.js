import React from 'react';
import {renderWithProviders} from './utils_for_tests';
import {NavigationContainer} from '@react-navigation/native';
import {ConvertScreen} from '../screens/convert';
import {screen} from '@testing-library/react-native';

const mockedNavigate = jest.fn();
const mockedgoBack = jest.fn();

jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate,
      goBack: mockedgoBack,
    }),
    useRoute: () => ({
      params: {
        cryptoValue: '0',
      },
    }),
  };
});

describe('ConvertScreen', () => {
  let component;
  beforeEach(() => {
    component = renderWithProviders(
      <NavigationContainer>
        <ConvertScreen />
      </NavigationContainer>,
    );
  });

  it('snapshot of screen to JSON', () => {
    const {toJSON} = renderWithProviders(
      <NavigationContainer>
        <ConvertScreen />
      </NavigationContainer>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly convert screen', () => {
    expect(component).toBeTruthy();
  });

  it('renders correctly the safe Area view', () => {
    const item = screen.getByTestId('safeAreaViewTest');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the text title convert screen', () => {
    const item = screen.getByTestId('titleConvertScreenTest');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the view text input', () => {
    const item = screen.getByTestId('viewTextInputTest');
    expect(item).toBeOnTheScreen();
  });
  it('renders correctly the second view text input', () => {
    const item = screen.getByTestId('viewTextInput2Test');
    expect(item).toBeOnTheScreen();
  });
  it('renders correctly the select picker up', () => {
    const item = screen.getByTestId('selectPickerUp');
    expect(item).toBeOnTheScreen();
  });

  it('renders correctly the select picker down', () => {
    const item = screen.getByTestId('selectPickerDown');
    expect(item).toBeOnTheScreen();
  });
});
