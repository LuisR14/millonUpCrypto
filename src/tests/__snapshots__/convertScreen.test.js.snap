// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConvertScreen snapshot of screen to JSON 1`] = `
<RCTSafeAreaView
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "marginHorizontal": 20,
    }
  }
  testID="safeAreaViewTest"
>
  <Text
    style={
      {
        "color": "black",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginTop": 20,
        "paddingHorizontal": 10,
      }
    }
    testID="titleConvertScreenTest"
  >
    Convert
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderBottomLeftRadius": 8,
        "borderRadius": 8,
        "elevation": 10,
        "flexDirection": "row",
        "height": 115,
        "marginTop": 20,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
        "shadowOffset": {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 5,
        "width": 660,
        "zIndex": 1,
      }
    }
    testID="viewTextInputTest"
  >
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "zIndex": 5000,
          },
          {
            "width": 280,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "lightgray",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 3,
            "width": "100%",
            "zIndex": 5000,
          }
        }
        testID="selectPickerUp"
      >
        <Text
          style={
            [
              {
                "color": "#000",
                "flex": 1,
              },
              {
                "color": "black",
              },
              false,
              {},
            ]
          }
        >
          Select
        </Text>
        <View
          style={
            [
              {
                "marginLeft": 10,
              },
              {},
            ]
          }
        >
          <Image
            source={
              {
                "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
              }
            }
            style={
              [
                {
                  "height": 20,
                  "width": 20,
                },
                {},
              ]
            }
          />
        </View>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        presentationStyle="fullScreen"
        visible={false}
      />
    </View>
    <TextInput
      editable={true}
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="0.00"
      placeholderTextColor="black"
      returnKeyType="done"
      style={
        {
          "color": "black",
          "flex": 1,
          "fontSize": 18,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      {
        "marginBottom": 40,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
      }
    }
  />
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <svg
      data-file-name="SvgChange"
      fill="black"
    />
  </View>
  <View
    style={
      {
        "marginBottom": 20,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderBottomLeftRadius": 8,
        "borderRadius": 8,
        "elevation": 10,
        "flexDirection": "row",
        "height": 115,
        "marginTop": 20,
        "paddingHorizontal": 10,
        "paddingVertical": 10,
        "shadowOffset": {
          "height": 10,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 5,
        "width": 660,
      }
    }
    testID="viewTextInput2Test"
  >
    <View
      style={
        [
          {
            "width": "100%",
          },
          {
            "zIndex": 5000,
          },
          {
            "width": 280,
          },
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "lightgray",
            "borderColor": "transparent",
            "borderRadius": 8,
            "borderWidth": 1,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "minHeight": 50,
            "opacity": 1,
            "paddingHorizontal": 10,
            "paddingVertical": 3,
            "width": "100%",
            "zIndex": 5000,
          }
        }
        testID="selectPickerDown"
      >
        <View
          style={
            [
              {
                "marginRight": 10,
              },
              {},
            ]
          }
        >
          <Image
            source={
              {
                "uri": "https://coinicons-api.vercel.app/api/icon/usd",
              }
            }
            style={
              {
                "height": 40,
                "width": 40,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "color": "#000",
                "flex": 1,
              },
              {
                "color": "black",
              },
              {},
              false,
            ]
          }
        >
          USD
        </Text>
        <View
          style={
            [
              {
                "marginLeft": 10,
              },
              {},
            ]
          }
        >
          <Image
            source={
              {
                "testUri": "../../react-native-dropdown-picker/src/themes/light/icons/arrow-down.png",
              }
            }
            style={
              [
                {
                  "height": 20,
                  "width": 20,
                },
                {},
              ]
            }
          />
        </View>
      </View>
      <Modal
        animationType="slide"
        hardwareAccelerated={false}
        onRequestClose={[Function]}
        presentationStyle="fullScreen"
        visible={false}
      />
    </View>
    <TextInput
      editable={false}
      keyboardType="decimal-pad"
      onChangeText={[Function]}
      placeholder="0.00"
      placeholderTextColor="black"
      returnKeyType="done"
      style={
        {
          "color": "black",
          "flex": 1,
          "fontSize": 18,
          "paddingHorizontal": 10,
          "paddingVertical": 5,
        }
      }
      value="0.00"
    />
  </View>
</RCTSafeAreaView>
`;
